src/app/model.ts

```diff
@@ -1,36 +1,128 @@
-import {Tickets, TicketType, Priority} from 'node-zendesk';
+import {Tickets} from 'node-zendesk';
 
-import {configStore} from '../store';
+import {AppFormValue, AppFormValues} from 'mattermost-redux/types/apps';
 
-type CreateTicketFormValues = {
-    subject: string;
-    type: TicketType;
-    priority: Priority;
-    additional_message: string;
-    post_message: string;
-}
+import {fieldNames, fieldValidation, getMultiselectValues} from '../utils';
 
-export const getTicketFromForm = (values: CreateTicketFormValues): Tickets.CreatePayload => {
-    const mmSignature = '*message created from Mattermost message.*\n' + configStore.getSiteURL();
+import {configStore} from '../store';
 
-    const additionalMessage = values.additional_message || '';
-    const postMessage = values.post_message || '';
+export type fieldValidationErrors = Record<string, string> | {};
 
-    const zdMessage = additionalMessage + '\n' +
-            postMessage + '\n' +
-            mmSignature;
+interface IticketFromFrom {
+    getTicket(): Tickets.CreatePayload;
+    fieldValidationErrors: fieldValidationErrors;
+}
 
-    const ticket: Tickets.CreatePayload = {
-        ticket: {
-            subject: values.subject,
-            type: values.type,
-            priority: values.priority,
+export class TicketFromForm implements IticketFromFrom {
+    formValues: AppFormValues;
+    ticket: Tickets.CreateModel;
+    fieldValidationErrors: any
+
+    constructor(values: AppFormValues) {
+        this.formValues = values;
+        this.fieldValidationErrors = {};
+        this.ticket = {
             comment: {
-                body: zdMessage,
+                body: this.getPostMessage(),
             },
-        },
-    };
-    return ticket;
-};
+        };
+    }
+
+    getPostMessage(): string {
+        const mmSignature = '*message created from Mattermost message.*\n' + configStore.getSiteURL();
+
+        const additionalMessage = this.formValues[fieldNames.additionalMessage] || '';
+        const postMessage = this.formValues[fieldNames.postMessage] || '';
+
+        const zdMessage = additionalMessage + '\n' +
+                postMessage + '\n' +
+                mmSignature;
+
+        return zdMessage;
+    }
+
+    getTicket(): Tickets.CreatePayload {
+        return this.mapFormValuesToTicket(this.ticket);
+    }
+
+    // mapFormValuesToTicket returns a zendesk ticket with AppCall values mapped
+    // 1:1 to a zendesk ticket
+    mapFormValuesToTicket(ticket: Tickets.CreateModel): Tickets.CreatePayload {
```

Comment:
I'm actually 5/5 on this one.  I had initially created this with functions but specifically converted to a class so that I could stop passing around the state to functions. I just missed removing modifying this function signature to remove the ticket input and use this.ticket instead.  Thanks for catching this one.